pkg/pillar/cmd/zedmanager/handleresolve.go

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) 2017-2018 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package zedmanager
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	log "github.com/sirupsen/logrus"
+)
+
+// MaybeAddResolveConfig will look for the container images for which
+// we need to resolve tags while purging the app instances
+func MaybeAddResolveConfig(ctx *zedmanagerContext, config types.AppInstanceConfig,
+	status *types.AppInstanceStatus) bool {
+
+	log.Infof("MaybeAddResolveConfig for %s", status.Key())
+	changed := false
+	for _, sc := range config.StorageConfigList {
+		ss := lookupStorageStatus(status, sc)
+		if ss != nil {
+			continue
+		}
+		if status.PurgeInprogress == types.NotInprogress {
```

Comment:
I understand why you are doing this, and I didn't think about it earlier.

But it might be easier to put another boolean in StorageStatus called ResolveDone, which is set once you get an answer from the resolver (and update the sha).

That way doUpdate can walk all of the StorageStatus and see whether it needs to resolve something, and if so call a MaybeAddResolveConfig for a single StorageStatus and not for the complete app instance.

Reply:
It makes sense to me. I will update it.